rpyc/core/protocol.py

```diff
@@ -560,6 +560,15 @@ def _handle_setattr(self, oid, name, value):
         return self._access_attr(oid, name, (value,), "_rpyc_setattr", "allow_setattr", setattr)
     def _handle_callattr(self, oid, name, args, kwargs):
         return self._handle_getattr(oid, name)(*args, **dict(kwargs))
+    def _handle_ctxexit(self, oid, exc):
+        if exc:
```

Comment:
What's the reason for the `try...except` here? To support context managers that require real traceback object?

The traceback generated here is not the one corresponding to the original exception, so I'm almost thinking it might be better to let context managers fail that require a traceback. I'd tend to remove the whole `if...else` block and simply pass `self._handle_getattr(oid, "__exit__")(exc, type(exc), tb)` instead. CMs that require tracebacks are probably more of an edge-case anyway. What do you think?